src/index.ts

```diff
@@ -0,0 +1,25 @@
+import { IPinfoWrapper } from "node-ipinfo";
+import defaultIPSelector from "./ip-selector/default-ip-selector";
+import originatingIPSelector from "./ip-selector/originating-ip-selector";
+
+type MiddlewareOptions = {
+    token?: string;
+    cache?: any;
+    timeout?: number;
+    ipSelector?: (req: any) => string;
+};
+  
+const ipInfoMiddleware = ({ token = "", cache, timeout, ipSelector }: MiddlewareOptions = {}) => {
+    const ipinfo = new IPinfoWrapper(token, cache, timeout);
+  if (ipSelector == null || typeof ipSelector !== "function") {
```

Comment:
The spacing is messed up here. Maybe we can add `prettier` and stabilize things. Then we can make spacing 4 as well.